### PR TITLE
[DO NOT MERGE] Add hide_chapter_nav option to guides

### PR DESCRIPTION
**Pull request by andysellick**
- will be used to control the display of contents list and other guide elements where the guide is inside a step by step navigation

Trello card: https://trello.com/c/RNsA7E4Q/763-hide-chapter-navigation-on-mainstream-guides

### CODE REVIEW

File: examples/guide/frontend/guide-with-step-navs.json [hunk 1]
```diff
@@ -886,7 +886,8 @@
...
+        "hide_chapter_navigation": true
```
[the remark this answers is not in the document]
Comment (andysellick) [reply examples/guide/frontend/guide-with-step-navs.json]: Any help would be appreciated. Is it clear what I'm trying to do? I need to add the `hide_chapter_navigation` property as a boolean on parts. Had a real headache getting it inserted in with a `$ref`.

[the remark this answers is not in the document]
Comment (cbaines) [reply examples/guide/frontend/guide-with-step-navs.json]: I've tested this locally, if you force the `parts` to be an object, then the validation fails:
```
 Error validating example examples/guide/publisher_v2/guide.json: The property '#/details/parts' of type array did not match the following type: object
```

File: dist/formats/guide/frontend/schema.json [hunk 2]
```diff
@@ -320,7 +320,15 @@
...
-          "$ref": "#/definitions/parts"
+          "properties": {
+            "hide_chapter_navigation": {
+              "description": "Hide guide elements if this part is part of a step by step navigation",
+              "type": "boolean"
+            },
+            "parts": {
+              "$ref": "#/definitions/parts"
+            }
+          }
```
[the remark this answers is not in the document]
Comment (cbaines) [reply dist/formats/guide/frontend/schema.json]: Just talked to Leena in the office as well, I'd suggest adding the `hide_chapter_navigation` bit to the `parts` thing, as an optional (so not in `required`) field. At the moment, the reason I think that the example is passing validation, is that nothing within the parts array is being validated, as these properties are not relevant, and the type is unspecified.

File: examples/guide/frontend/guide-with-step-navs.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (andysellick) [reply examples/guide/frontend/guide-with-step-navs.json]: Have updated as per your suggestion, thanks @cbaines. Can you check I've got it right now?

[the remark this answers is not in the document]
Comment (cbaines) [reply examples/guide/frontend/guide-with-step-navs.json]: Yep, that looks like it'll work :+1:

**Comment by andysellick** [comment]
Superceded by #804, changing from a per-part option to a whole-guide option.

**Pull request closed without merging**